crates/utils/src/logging.rs

```diff
@@ -57,6 +57,27 @@ pub fn setup_tracing(otel: OpenTelemetry) -> Result<()> {
     tracing::subscriber::set_global_default(subscriber).map_err(Into::into)
 }
 
+#[cfg(feature = "testing")]
+pub fn setup_test_tracing() -> Result<(
+    tokio::sync::mpsc::UnboundedReceiver<opentelemetry_sdk::trace::SpanData>,
+    tokio::sync::mpsc::UnboundedReceiver<()>,
+)> {
+    static INIT: std::sync::Once = std::sync::Once::new();
+    INIT.call_once(|| {
+        opentelemetry::global::set_text_map_propagator(TraceContextPropagator::new());
+    });
```

Comment:
@Mirko-von-Leipzig my understanding of your point from the issue around global state leads me to believe this is an appropriate solution. Every test that wants to validate traces needs to call this function in any case.

> The main issue iiuc is that the exporter is registered globally which makes running such tests.. a bit difficult.

The only problem would be if we had parallel tests that ran `setup_tracing()`. But I don't think that will ever be the case.

LMK if I misunderstood anything.

Reply:
> The only problem would be if we had parallel tests that ran setup_tracing(). But I don't think that will ever be the case.

What prevents them from running tests concurrently? By default `cargo test` would run tests concurrently, so I imagine this function call would somehow have to block until the previous test is completed its usage of the span exporter?

Or maybe I'm missing something :)

Reply:
Yea there is another global lower in that function that I didn't notice. Looks like the tests will indeed clobber. Think best thing is to force serialization of the trace tests. Used this in the past https://crates.io/crates/serial_test

Reply:
Ah nice yeah that looks like a good solution! I like that it even let's you group serial tests together.